### DISCUSSION TITLE
Issue with Clobberall

### DISCUSSION THREAD

**Discussion by sreihani** [idaholab/moose#28625] [Q&A Meshing]
I guess "Make clobberall" does have an impact on upstream of MOOSE setup. I updated MOOSE repository and all pets, libmesh and Wasp. However, after clobbering, I was forced to update wasp again and consequently wasp itself failed! I am not sure what the output of a clobberall is supposed to be, but my understanding is that should not touch those libraries already updated. I am wondering if this step can be skipped, if so any consequences? FYI, I built moose from source.

**Comment by GiudGiud** [comment]
Hello

We do expect make clobberall to clear local dependencies.
However we don't expect trouble re installing after. 
Did you update before clobbering ?

Can you use git clean -Xfd to have a deeper clean

**Reply by milljm** [reply]
I tend to stay away from `clobberall` myself. It does things we have programmatically set in a Makefile. And I am not good at reading Makefiles, and understanding what it is I am reading in them.

But when I run the following git clean commands _I know what it is I am cleaning. And I like to know when things are being cleaned_.

```bash
cd moose
git clean -xfd   # cleans moose/* (everything that is not a submodule)
git submodule foreach --recursive git clean -xfd  # cleans all submodules, and their submodules
```
It was only recently that I was shown the `-X` being safer than what I've been using for years (`-x`). I still prefer `-x`, as I like to end up in a pristine repo (see `git clean --help` if curious).

Basically with the above two commands, I can manually build PETSc, libMesh, and WASP and then run `git clean -xfd` at the root of the moose repo, and not have to worry about destroying PETSc, libMesh, or WASP.
